metrics/metrics.go

```diff
@@ -84,21 +83,20 @@ func RegisterRootScope(rep ScopeFunc) {
 		panic("There can be only one metrics reporter")
 	}
 
-	_scopeFunc = rep
+	_scopeFunc = scopeFunc
 }
 
-// RootScope returns the provided stats reporter, or nil
-func RootScope(i ScopeInit) (tally.Scope, io.Closer) {
+// RootScope returns the provided metrics scope and stats reporter, or nil if not provided
+func RootScope(i ScopeInit) (tally.Scope, tally.StatsReporter, io.Closer) {
 	_mu.Lock()
 	defer _mu.Unlock()
 
 	if _scopeFunc != nil {
-		rep, closer, err := _scopeFunc(i)
+		scope, reporter, closer, err := _scopeFunc(i)
 		if err != nil {
 			panic(fmt.Sprintf("Failed to initialize metrics reporter %v", err))
 		}
-		return rep, closer
+		return scope, reporter, closer
 	}
-
-	return nil, nil
+	return nil, nil, nil
```

Comment:
I don't think a scope registry is necessary here. This exchange happens literally in one place and the caller (service.go) assigns it to a struct. Creating a new struct seems redundant. Go has strongly typed multiple return values, so the caller knows exactly what to expect. 
With errors, if this is not a panic, I completely agree. There is no use for the returned error if we are handling it within.